### PR TITLE
Feature/map 97/add location

### PR DESCRIPTION
**Pull request by Yojuks**
Now work only functional:
- when you click on the button to add the location you should click on the map and choose location coordinates - and after that you will see a popup where you can send name, description, and images 
this task is not completed but this part needs to other coleags

### CODE REVIEW

**Review by romawht22 — APPROVED** [review]

File: frontend/src/components/ChildrenBigPopupLocation/ChildrenBigPopupLocation.tsx [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+
+      {/* <img
+        src=""
+        alt=""
+        style={{
+          width: '200px',
+          height: '200px',
+          backgroundColor: 'white'
+        }}
+      /> */}
```
Comment (obozhko-dev) [inline frontend/src/components/ChildrenBigPopupLocation/ChildrenBigPopupLocation.tsx]: delete, if this code not use

File: frontend/src/components/ChildrenBigPopupLocation/ChildrenBigPopupLocation.tsx [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+        placeholder="Minimum 3 rows"
+        style={{ width: 200 }}
+      />
+
+      <Autocomplete
+        multiple
+        id="tags-outlined"
+        options={getFiltersForUser()}
+        getOptionLabel={option => option}
+        // defaultValue={[autocompleteFilters[0]]}
```
Comment (obozhko-dev) [inline frontend/src/components/ChildrenBigPopupLocation/ChildrenBigPopupLocation.tsx]: delete, if this code not use

File: frontend/src/components/ChildrenBigPopupLocation/ChildrenBigPopupLocation.tsx [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+import axios from 'axios';
+import React, { useState, useRef } from 'react';
+import { latlngType } from '../../../types';
+import { getFiltersForUser } from '../../static/mainFIlters';
+
+type Props = {
+  coordinate: latlngType;
+};
+
+const ChildrenBigPopupLocation = ({ coordinate }: Props) => {
```
Comment (obozhko-dev) [inline frontend/src/components/ChildrenBigPopupLocation/ChildrenBigPopupLocation.tsx]: maybe you can change component name? maybe something like "NewLocation" or "CreateLocation", "AddLocation"

File: frontend/src/static/mainFIlters.tsx [hunk 4]
```diff
@@ -1,12 +1,28 @@
...
     id: 1,
+    forLoggedUser: false,
     type: 'Cost',
     values: ['free', 'low cost', 'high cost']
   },
   {
     id: 2,
+    forLoggedUser: false,
     type: 'Seasonal',
+    // cтворити форму з фільтрами - де я маю mainFilters обєднати перших 2 values прочекати його і додати в автокомліт
```
Comment (obozhko-dev) [inline frontend/src/static/mainFIlters.tsx]: delete, if this code not use

**Review by obozhko-dev — APPROVED** [review]

File: backend/controllers/LocationsController.ts [hunk 5]
```diff
@@ -160,7 +171,7 @@ const LocationsController = {
 
   async postPersonalLocation(req: Request, res: Response) {
     try {
-      const { locationName, description, coordinates } = req.body;
+      const { locationName, description, coordinates, filters } = req.body;
```
Comment (6mahdapihka9) [inline backend/controllers/LocationsController.ts:144]: change type of coordinates from [string,string] to [number, number]

**Review by 6mahdapihka9 — CHANGES_REQUESTED** [review]

File: frontend/src/components/ChildrenBigPopupLocation/ChildrenBigPopupLocation.tsx (hunk 1 above)
Comment (Yojuks) [reply frontend/src/components/ChildrenBigPopupLocation/ChildrenBigPopupLocation.tsx]: we should discuss about future functional with that commented code

File: frontend/src/static/mainFIlters.tsx (hunk 4 above)
Comment (Yojuks) [reply frontend/src/static/mainFIlters.tsx]: done

File: backend/controllers/LocationsController.ts (hunk 5 above)
Comment (Yojuks) [reply backend/controllers/LocationsController.ts:144]: done

File: frontend/src/components/ChildrenBigPopupLocation/ChildrenBigPopupLocation.tsx (hunk 2 above)
Comment (Yojuks) [reply frontend/src/components/ChildrenBigPopupLocation/ChildrenBigPopupLocation.tsx]: done

File: frontend/src/components/ChildrenBigPopupLocation/ChildrenBigPopupLocation.tsx (hunk 3 above)
Comment (Yojuks) [reply frontend/src/components/ChildrenBigPopupLocation/ChildrenBigPopupLocation.tsx]: Done

File: backend/controllers/LocationsController.ts [hunk 6]
```diff
@@ -180,15 +160,15 @@ const LocationsController = {
...
           arrayPhotos: imageUrls,
           description: description,
           comments: [],
           rating: {
             likes: [],
             dislikes: []
           },
-          filters: [],
+          filters: filters.split(','),
           author: _id
```
Comment (6mahdapihka9) [inline backend/controllers/LocationsController.ts:172]: have you tested the case with empty filters?

and what do you receive from frontend when you don't set any filter?

File: backend/routes/index.ts [hunk 7]
```diff
@@ -22,16 +24,21 @@ router.post('/signup', AuthController.signUp);
...
+router.get(
+  '/subscriptions',
+  passport.authenticate('jwt', { session: false }),
+  SubscriptionsController.getSubscriptions
+);
+
+router.post('/signup', AuthController.signUp);
+router.post('/signin', AuthController.signIn);
+router.post('/forgot-password', AuthController.forgotPassword);
+
```
Comment (6mahdapihka9) [inline backend/routes/index.ts]: duplicated rows

File: frontend/src/components/CreateLocation/CreateLocation.tsx [hunk 8]
```diff
@@ -0,0 +1,173 @@
...
+
+      {/* <img
+        src=""
+        alt=""
+        style={{
+          width: '200px',
+          height: '200px',
+          backgroundColor: 'white'
+        }}
+      /> */}
```
Comment (6mahdapihka9) [inline frontend/src/components/CreateLocation/CreateLocation.tsx:122]: should be removed before merge

File: frontend/types.ts [hunk 9]
```diff
@@ -53,3 +53,14 @@ export type UserDataType = {
   subscribers: string[];
   subscriptions: string[];
 };
+
+export type LocationType = {
+  locationName: string;
```
Comment (6mahdapihka9) [inline frontend/types.ts:58]: name duplicated in row 17

Comment (6mahdapihka9) [reply frontend/types.ts:58]: later

File: frontend/src/components/CreateLocation/CreateLocation.tsx (hunk 8 above)
Comment (6mahdapihka9) [reply frontend/src/components/CreateLocation/CreateLocation.tsx:122]: later

File: backend/routes/index.ts (hunk 7 above)
Comment (6mahdapihka9) [reply backend/routes/index.ts]: now

File: backend/controllers/LocationsController.ts (hunk 6 above)
Comment (6mahdapihka9) [reply backend/controllers/LocationsController.ts:172]: later

**Review by 6mahdapihka9 — CHANGES_REQUESTED** [review]
I'd recommend to change the name of the endpoint to /location/create

later

File: frontend/src/components/CreateLocation/CreateLocation.tsx [hunk 10]
```diff
@@ -0,0 +1,173 @@
...
+  };
+
+  const handleFilesChange = (e: any) => {
+    const { files: filesLst } = e.currentTarget;
+    const filesListArr = [];
+
+    if (filesLst) {
+      for (let i = 0; i < filesLst.length; i += 1) {
+        filesListArr.push(filesLst[i]);
+      }
```
Comment (6mahdapihka9) [inline frontend/src/components/CreateLocation/CreateLocation.tsx:79]: just why? :)

**Review by 6mahdapihka9 — APPROVED** [review]

File: frontend/src/components/CreateLocation/CreateLocation.tsx (hunk 10 above)
Comment (Yojuks) [reply frontend/src/components/CreateLocation/CreateLocation.tsx:79]: this code i copy and paste from Yuriy  - this is need for images

**Pull request merged**